File: admin/src/types/promoBanner.ts
export type PromoBannerLinkType = 'category' | 'subcategory' | 'product_type' | 'product' | 'custom_url';
export type PromoBannerLayoutType = 'small' | 'large';

export interface PromoBannerItem {
  id: string;
  internal_name: string;
  layout_type: PromoBannerLayoutType;
  label_en: string;
  label_ar: string;
  title_en: string;
  title_ar: string;
  cta_text_en: string;
  cta_text_ar: string;
  desktop_image_url: string;
  mobile_image_url: string;
  alt_text: string;
  link_type: PromoBannerLinkType;
  category_id: string;
  subcategory_id: string;
  product_type_id: string;
  product_id: string;
  custom_url: string;
  sort_order: number;
  is_active: boolean;
  is_published: boolean;
  created_at: string;
  updated_at: string;
}

export const INITIAL_PROMO_BANNER: Omit<PromoBannerItem, 'id' | 'created_at' | 'updated_at'> = {
  internal_name: '',
  layout_type: 'small',
  label_en: '',
  label_ar: '',
  title_en: '',
  title_ar: '',
  cta_text_en: 'Shop now',
  cta_text_ar: 'تسوق الآن',
  desktop_image_url: '',
  mobile_image_url: '',
  alt_text: '',
  link_type: 'category',
  category_id: '',
  subcategory_id: '',
  product_type_id: '',
  product_id: '',
  custom_url: '',
  sort_order: 0,
  is_active: true,
  is_published: false,
};

File: backend/Mayar.Api/Entities/PromoBanner.cs
using System;
using Mayar.Api.Helpers;

namespace Mayar.Api.Entities;

public class PromoBanner
{
    public Guid Id { get; set; }

    // Admin identification
    public string InternalName { get; set; } = string.Empty;

    // Layout type: "small" or "large"
    public string LayoutType { get; set; } = "small";

    // Label/Eyebrow text (bilingual)
    public string? LabelEnglish { get; set; }
    public string? LabelArabic { get; set; }

    // Main title (bilingual)
    public string? TitleEnglish { get; set; }
    public string? TitleArabic { get; set; }

    // CTA/Button text (bilingual)
    public string? CtaTextEnglish { get; set; }
    public string? CtaTextArabic { get; set; }

    // Media - Desktop and Mobile images
    public string? DesktopImageUrl { get; set; }
    public string? MobileImageUrl { get; set; }
    public string? AltText { get; set; }

    // Link configuration
    public string LinkType { get; set; } = "category"; // category, subcategory, product_type, product, custom_url
    public Guid? CategoryId { get; set; }
    public Guid? SubcategoryId { get; set; }
    public Guid? ProductTypeId { get; set; }
    public Guid? ProductId { get; set; }
    public string? CustomUrl { get; set; }

    // Navigation properties for link relationships
    public TopCategory? Category { get; set; }
    public MiddleCategory? Subcategory { get; set; }
    public BottomCategory? ProductType { get; set; }
    public Product? Product { get; set; }

    // Ordering and status
    public int SortOrder { get; set; } = 0;
    public bool IsActive { get; set; } = true;
    public bool IsPublished { get; set; } = false;

    // Timestamps
    public DateTime CreatedAt { get; set; } = DateTimeHelper.GetLocalTime();
    public DateTime UpdatedAt { get; set; } = DateTimeHelper.GetLocalTime();
}
